lib/bypass/uri.rb

```diff
@@ -3,7 +3,17 @@
 module Bypass
   class URI < Addressable::URI
     def append_to_query_values(params = {})
-      self.query_values = (query_values || {}).merge(params)
+      new_query_values = [].tap do |pairs|
+        params.each do |key, value|
+          pairs << "#{key}=#{value}"
+        end
+      end.join('&')
+
+      if self.query.nil?
+        self.query = "#{new_query_values}"
+      else
+        self.query += "&#{new_query_values}"
+      end
```

Comment:
I think this would be equivalent:

```ruby
pairs = params.keys.map { |key| "#{key}=#{params[key]}" }
pairs.unshift(query) if query
self.query = pairs.join("&")
```

Outstanding question: do we need to perform encoding on keys and values?